### PR TITLE
Increase fs.aio-max-nr on CI worker nodes

### PR DESCRIPTION
**Pull request by rzetelskik**
**Description of your changes:** As discovered during the debugging of https://github.com/scylladb/scylla-operator/issues/1231, our CI suffers from too few allowable concurrent requests with the current parallelism of the e2e suite. Until https://github.com/scylladb/scylla-operator/issues/749 is completed, this PR deploys a DaemonSet on worker nodes of our CI to increase the `fs.aio-max-nr` as a workaround.

**Which issue is resolved by this Pull Request:**
Resolves https://github.com/scylladb/scylla-operator/issues/1231

### CODE REVIEW

File: hack/.ci/run-e2e-gke.sh [hunk 1]
```diff
@@ -60,6 +60,45 @@ kubectl_create -f - <<EOF
...
+          command:
+          - /usr/bin/bash
+          - -euExo
+          - pipefail
+          - -O
+          - inherit_errexit
+          - -c
+          args:
+          - |
+            sysctl fs.aio-max-nr=0xffffffff
```
[the remark this answers is not in the document]
Comment (tnozicka) [reply hack/.ci/run-e2e-gke.sh:92]: Thanks, our CI clusters are not in the guaranteed class nor receive any significant traffic during the run, so the idea was more about fitting as many as we can without being artificially limited on this.

Does it have a relation to CPU count or the size of the cluster in general? When we recommend tuning this properly with https://github.com/scylladb/scylla-operator/issues/749 we need some metric to determine the value for the node while we don't know how many or how big clusters will run there. OR there are more advance options like advertising it as hardware resource using device plugins https://kubernetes.io/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins/ but that may be an overkill, at least at this point.

This PR is just about unstucking the CI that runs in the best effort QoS class / overcommitted.

[the remark this answers is not in the document]
Comment (rzetelskik) [reply hack/.ci/run-e2e-gke.sh:92]: Yup, that's just a workaround aimed at eliminating flakes until it's supported by NodeConfig - see https://github.com/scylladb/scylla-operator/issues/1231. Hence an over the top value of max unsigned long.

[the remark this answers is not in the document]
Comment (tnozicka) [reply hack/.ci/run-e2e-gke.sh:92]: might worth to put the reasons for this into a comment

File: hack/.ci/run-e2e-gke.sh [hunk 2]
```diff
@@ -60,6 +60,46 @@ kubectl_create -f - <<EOF
...
+
+            sleep infinity &
+            wait
```
Comment (tnozicka) [inline hack/.ci/run-e2e-gke.sh:96]: how is this different from `sleep infinity` on the foreground

File: hack/.ci/run-e2e-gke.sh [hunk 3]
```diff
@@ -60,6 +60,46 @@ kubectl_create -f - <<EOF
...
+        nodeSelector:
+          kubernetes.io/os: linux
+          pool: workers
```
Comment (tnozicka) [inline hack/.ci/run-e2e-gke.sh:99]: I think `"scylla.scylladb.com/node-type": "scylla"` would be more fitting and stable

Comment (rzetelskik) [reply hack/.ci/run-e2e-gke.sh:99]: I did that initially but changed to `pool: workers` to align with existing manifests: https://github.com/scylladb/scylla-operator/blob/25735c38e04ee9deb0fa5f471f2ce7951aff5b16/hack/.ci/manifests/namespaces/local-csi-driver/50_daemonset.yaml#L19. Does `pool: workers` have a different meaning, i.e. is worker pool potentially a superset of scylla node pool? Anyway `"scylla.scylladb.com/node-type": "scylla"` is neater so I'll change it.

File: hack/.ci/run-e2e-gke.sh (hunk 2 above)
Comment (rzetelskik) [reply hack/.ci/run-e2e-gke.sh:96]: sigint handling?

File: hack/.ci/run-e2e-gke.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (rzetelskik) [reply hack/.ci/run-e2e-gke.sh:93]: Added a PR description, a FIXME in the script and created a tracking issue. Let me know if that suffices.

File: hack/.ci/run-e2e-gke.sh (hunk 3 above)
Comment (tnozicka) [reply hack/.ci/run-e2e-gke.sh]: thanks. not really, but `scylla.scylladb.com/node-type` is more standard and we use it in our docs. `pool: workers` for probably for some convenience and could go away.

File: hack/.ci/run-e2e-gke.sh (hunk 2 above)
Comment (tnozicka) [reply hack/.ci/run-e2e-gke.sh:96]: interesting, sending `SIGINT` to `sleep` works for me
```
 tnozicka  ~  sleep infinity
^C
 tnozicka  ~  SIGINT  
```

Comment (rzetelskik) [reply hack/.ci/run-e2e-gke.sh:96]: Doesn't ^C send a SIGINT to the entire process group?
`kill -s INT -$PGID` as opposed to `kill -s INT $PID`

Comment (rzetelskik) [reply hack/.ci/run-e2e-gke.sh:96]: Shouldn't you kill the bash process in that case?

```bash
~ cat script.sh
#!/bin/bash
sleep infinity
~ ./script.sh &
~ ps o pid,ppid,pgid,comm
   6467    6289    6467 zsh
  13292    6467   13292 script.sh
  13294   13292   13292 sleep
  13500    6467   13500 ps
~ kill -s INT 13292
~ ps o pid,ppid,pgid,comm
   6467    6289    6467 zsh
  13292    6467   13292 script.sh
  13294   13292   13292 sleep
  13931    6467   13931 ps
~ kill -s INT -13292
[1]  + 13292 interrupt  ./script.sh
```

**Pull request merged**